### PR TITLE
PUB-1203 Add filter functionality to SJP press list

### PR DESCRIPTION
**Pull request by KianKwa**
### JIRA link

https://tools.hmcts.net/jira/browse/PUB-1203

### Change description

Add filter functionality to SJP press list

### Filter hidden:
image

### **Filter displayed:**
image

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/service/listManipulation/SjpPressListService.ts [hunk 1]
```diff
@@ -1,39 +1,99 @@
...
-                                    ).toFormat('d MMMM yyyy');
-                                }
-                            });
-
-                            hearing['offence'].forEach(offence => {
-                                const reportingRestriction = offence['reportingRestriction'].toString();
-                                offence['formattedReportingRestriction'] =
-                                    reportingRestriction.charAt(0).toUpperCase() + reportingRestriction.slice(1);
-                            });
+                            this.buildCases(hearing, rows);
```
Comment (furnivall) [inline src/main/service/listManipulation/SjpPressListService.ts:19]: nice

File: src/main/service/sjpFilterService.ts [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+        return filterStructure;
+    }
+
+    /**
+     * This method filters the cases for the SJP list based on the user selected options
+     * @param allCases The cases to filter.
+     * @param filterOptions The options that have been selected
+     * @private
+     */
+    private filterCases(allCases, filterOptions) {
```
Comment (furnivall) [inline src/main/service/sjpFilterService.ts:82]: Sorry - any chance this method could be broken down into submethods? A bit too much functionality in here

File: src/test/unit/mocks/SJPMockPage.json [hunk 3]
```diff
@@ -17,22 +17,24 @@
...
+                                                                "town": "Swansea",
+                                                                "postCode": "SA1 1AA"
                                                             }
                                                         },
                                                         "organisationDetails": {
                                                             "organisationName": "qU8QlEo@"
                                                         }
                                                     },
                                                     {
-                                                        "partyRole": "Prosecuter",
```
Comment (furnivall) [inline src/test/unit/mocks/SJPMockPage.json:35]: Thank you so much for doing this

File: src/test/unit/service/listManipulation/sjpPressListService.test.ts [hunk 4]
```diff
@@ -7,31 +7,49 @@ const rawSJPData = fs.readFileSync(path.resolve(__dirname, '../../mocks/SJPMockP
...
         const data = await sjpPressListService.formatSJPPressList(rawSJPData);
-        expect(data['courtLists'].length).to.equal(1);
+        expect(data.length).to.equal(2);
     });
 
-    it('should formatted date of birth in correct format', async () => {
+    it('should return accused name', async () => {
         const data = await sjpPressListService.formatSJPPressList(rawSJPData);
-        expect(
-            data['courtLists'][0]['courtHouse']['courtRoom'][0]['session'][0]['sittings'][0]['hearing'][0]['party'][0][
```
Comment (furnivall) [inline src/test/unit/service/listManipulation/sjpPressListService.test.ts:18]: this is so much sexier now

File: src/test/unit/mocks/SJPMockPage.json (hunk 3 above)
Comment (KianKwa) [reply src/test/unit/mocks/SJPMockPage.json:35]: lol know you would like that :)

**Review by joshblackmoor — APPROVED** [review]

File: src/main/service/sjpFilterService.ts (hunk 2 above)
Comment (KianKwa) [reply src/main/service/sjpFilterService.ts:82]: Have refactored the method a bit :)

**Review by furnivall — APPROVED** [review]

**Pull request closed without merging**